query/aql_batchexecutor.go

```diff
@@ -243,10 +243,12 @@ func (e *BatchExecutorImpl) reduce() {
 	memutils.WaitForCudaStream(e.stream, e.qc.Device)
 }
 
-func (e *BatchExecutorImpl) preExec(isLastBatch bool, start time.Time) {
+func (e *BatchExecutorImpl) preExec(isLastBatch bool, start time.Time, hasRegularFilter bool) {
 	e.isLastBatch = isLastBatch
 	// initialize index vector.
-	initIndexVector(e.qc.OOPK.currentBatch.indexVectorD.getPointer(), 0, e.qc.OOPK.currentBatch.size, e.stream, e.qc.Device)
+	if hasRegularFilter {
+		initIndexVector(e.qc.OOPK.currentBatch.indexVectorD.getPointer(), 0, e.qc.OOPK.currentBatch.size, e.stream, e.qc.Device)
```

Comment:
would interesting to know how will it work without the index vector

Reply:
for current batch, since there's no filter to evaluate, `initIndexVector`  will be skipped, `filter()` will actually do nothing.